### ISSUE TITLE
Scale up operation logs DeploymentFailed error for Azure VMAS-based Kubernetes 1.9.3 Cluster

### ISSUE THREAD

**Issue by jayanthpurushothaman** [kubernetes/autoscaler#703]
When attempting to scale up a K8S v1.9.3 Cluster (1 Master, 2 Nodes) with Azure Managed Disks _(K8S Cluster was deployed and upgraded using ACS Engine v0.13.0)_, the **cluster autoscaler** _(Docker image built from the **master** branch of the current code base as per [https://github.com/kubernetes/autoscaler/issues/533](url) and [https://github.com/kubernetes/autoscaler/issues/644](url))_ logs the following errors, even though the Scale Up operation seems to be successful and the Pods are scheduled.

I0310 15:01:54.964776       1 scale_up.go:195] Best option to resize: agentpool1
I0310 15:01:54.964796       1 scale_up.go:199] Estimated 1 nodes needed in agentpool1
I0310 15:01:54.964807       1 scale_up.go:288] Final scale-up plan: [{agentpool1 2->3 (max: 3)}]
I0310 15:01:54.964819       1 scale_up.go:340] Scale-up: setting group agentpool1 size to 3
I0310 15:01:54.996988       1 azure_agent_pool.go:209] Waiting for deploymentsClient.CreateOrUpdate(_ResourceGroupName_, cluster-autoscaler-761424867, {%!s(*resources.DeploymentProperties=&{0xc420cebd08 <nil> 0xc420cebd10 <nil> Incremental <nil>})})
W0310 15:09:56.391166       1 clusterstate.go:248] Disabling scale-up for node group agentpool1 until 2018-03-10 15:39:56.391154808 +0000 UTC m=+8749.309110125
**E0310 15:09:56.391219       1 static_autoscaler.go:302] Failed to scale up: failed to increase node group size: resources.DeploymentsClient#CreateOrUpdate: Failure sending request: StatusCode=200 -- Original Error: Long running operation terminated with status 'Failed': Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details."**
I0310 15:10:06.456531       1 azure_manager.go:242] Refreshed ASG list, next refresh after 2018-03-10 15:11:06.456517902 +0000 UTC m=+7019.374473118

Looking at the Deployments in Azure, the following message was found:

	"error": {
		"code": "DeploymentFailed",
		"details": [{
			"code": "Conflict",
			"message": "{\r\n  \"error\": {\r\n    \"code\": \"PropertyChangeNotAllowed\",\r\n    \"target\": \"dataDisk.createOption\",\r\n    \"message\": \"Changing property 'dataDisk.createOption' is not allowed.\"\r\n  }\r\n}"
		}],
		"message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details."
	}

Any feedback on this would be definitely helpful. Thanks.

**Comment by mwielgus** [comment]
cc: @feiskyer

**Comment by feiskyer** [comment]
@jayanthpurushothaman Have you changed data disk options after deployments?

**Comment by jayanthpurushothaman** [comment]
@feiskyer Not that I am aware of. The ACS cluster was deployed as 1.8.2 and upgraded to 1.9.3 using standard ACS Engine commands. In fact, we don't use additional disks at all, apart from whatever disks the ACS Engine deploys from our standard Cluster JSON definition.

**Comment by feiskyer** [comment]
@jayanthpurushothaman The problem may be caused by upgrading. Azure ARM templates changed after upgrading.

Let me see how to solve this problem.

/assign

**Comment by jayanthpurushothaman** [comment]
@feiskyer Ok. Thanks a lot. 

By the way, would ACS Engine **v0.13.0** allow direct deployments of Kubernetes v1.9.x? The reason why we deployed a v1.8.2 first was because the `acs-engine generate` command was throwing errors when we specify anything above v1.8.2 for `orchestratorVersion` parameter in the JSON file.

**Comment by feiskyer** [comment]
> By the way, would ACS Engine v0.13.0 allow direct deployments of Kubernetes v1.9.x? The reason why we deployed a v1.8.2 first was because the acs-engine generate command was throwing errors when we specify anything above v1.8.2 for orchestratorVersion parameter in the JSON file.

@jayanthpurushothaman  Yep, the lastest release version is v0.14.0, you can specify "orchestratorVersion" to "1.9.3" or "orchestratorRelease" to "1.9"

**Comment by feiskyer** [comment]
@mwielgus Since there are a lot of v1.9.X users, do we accept new features into autoscaler 1.1 branch?

**Comment by mwielgus** [comment]
Yes.

**Comment by mwielgus** [comment]
Or to be more precise, yes, if they are not too big.

**Comment by feiskyer** [comment]
@mwielgus For azure support, it may include the whole azure package and related vendors. Let me file a PR and check whether it's acceptable

**Comment by feiskyer** [comment]
> For azure support, it may include the whole azure package and related vendors. Let me file a PR and check whether it's acceptable

Tried last week, there'll be too many vendors upgrade.  Let's only keep 1.10+ support for Azure vmas.
